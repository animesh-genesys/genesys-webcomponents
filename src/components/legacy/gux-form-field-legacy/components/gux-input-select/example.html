<form onchange="notify(event)" oninput="notify(event)">
  <h1>gux-form-field-legacy (Select)</h1>

  <p>slot a select element into a gux-form-field-legacy element</p>

  <h2>Label Position Above</h2>

  <gux-form-field-legacy label-position="above">
    <select slot="input" name="select">
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
      <option value="option3">Option 3</option>
    </select>
    <label slot="label">Select</label>
  </gux-form-field-legacy>

  <gux-form-field-legacy label-position="above">
    <select slot="input" name="select">
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
      <option value="option3">Option 3</option>
    </select>
    <label slot="label">Select</label>
    <span slot="error">Required</span>
  </gux-form-field-legacy>

  <gux-form-field-legacy label-position="above">
    <select slot="input" name="select" disabled>
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
      <option value="option3">Option 3</option>
    </select>
    <label slot="label">Select</label>
  </gux-form-field-legacy>

  <h2>Label Position Beside</h2>

  <gux-form-field-legacy label-position="beside">
    <select slot="input" name="select">
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
      <option value="option3">Option 3</option>
    </select>
    <label slot="label">Select</label>
  </gux-form-field-legacy>

  <gux-form-field-legacy label-position="beside">
    <select slot="input" name="select">
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
      <option value="option3">Option 3</option>
    </select>
    <label slot="label">Select</label>
    <span slot="error">Required</span>
  </gux-form-field-legacy>

  <gux-form-field-legacy label-position="beside">
    <select slot="input" name="select" disabled>
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
      <option value="option3">Option 3</option>
    </select>
    <label slot="label">Select</label>
  </gux-form-field-legacy>

  <h2>With a smaller width</h2>
  <gux-form-field-legacy label-position="above" style="width: 120px">
    <select slot="input" name="select">
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
      <option value="option3">Option 3</option>
    </select>
    <label slot="label">Select</label>
  </gux-form-field-legacy>
</form>
