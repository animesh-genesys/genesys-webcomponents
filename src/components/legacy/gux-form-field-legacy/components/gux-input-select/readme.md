# gux-input-select

<!-- Auto Generated Below -->


## Slots

| Slot      | Description                      |
| --------- | -------------------------------- |
| `"input"` | Required slot for select element |


## Dependencies

### Used by

 - [gux-form-field-legacy](../..)

### Depends on

- [gux-icon](../../../../stable/gux-icon)

### Graph
```mermaid
graph TD;
  gux-input-select --> gux-icon
  gux-form-field-legacy --> gux-input-select
  style gux-input-select fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
