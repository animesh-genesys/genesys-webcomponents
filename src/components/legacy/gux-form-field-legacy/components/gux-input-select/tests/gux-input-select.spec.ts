import { newSpecPage, SpecPage } from '@stencil/core/testing';
import MutationObserver from 'mutation-observer';

import { GuxInputSelect } from '../gux-input-select';

const components = [GuxInputSelect];
const language = 'en';

describe('gux-input-select', () => {
  let page: SpecPage;

  beforeEach(async () => {
    global.MutationObserver = MutationObserver;

    page = await newSpecPage({
      components,
      html: `
        <gux-input-select>
          <select slot="input">
            <option value="option1">Option 1</option>
            <option value="option2">Option 2</option>
            <option value="option3">Option 3</option>
          </select>
        </<gux-input-select>
      `,
      language
    });
  });

  it('should build', async () => {
    expect(page.rootInstance).toBeInstanceOf(GuxInputSelect);
  });

  it('should render', async () => {
    expect(page.root).toMatchSnapshot();
  });
});
