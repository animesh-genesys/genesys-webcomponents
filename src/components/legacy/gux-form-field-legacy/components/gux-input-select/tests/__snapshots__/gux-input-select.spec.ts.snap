// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-input-select should render 1`] = `
<gux-input-select>
  <!---->
  <!--<gux-input-select-->
  <div class="gux-input-container">
    <select slot="input">
      <option value="option1">
        Option 1
      </option>
      <option value="option2">
        Option 2
      </option>
      <option value="option3">
        Option 3
      </option>
    </select>
    <gux-icon decorative="" iconname="ic-dropdown-arrow"></gux-icon>
  </div>
</gux-input-select>
`;
